src/prpy/planning/vectorfield.py

```diff
@@ -240,45 +427,46 @@ def FollowVectorField(self, robot, fn_vectorfield, fn_terminate,
 
         env = robot.GetEnv()
         active_indices = robot.GetActiveDOFIndices()
-        q_limit_min, q_limit_max = robot.GetActiveDOFLimits()
+
+        # Get the robot's joint velocity limits
         qdot_limit = robot.GetDOFVelocityLimits(active_indices)
 
+        # Create a new trajectory matching the current
+        # robot's joint configuration specification
         cspec = robot.GetActiveConfigurationSpecification('linear')
         cspec.AddDeltaTimeGroup()
         cspec.ResetGroupOffsets()
-
         path = RaveCreateTrajectory(env, '')
         path.Init(cspec)
 
         time_start = time.time()
 
         def fn_wrapper(t, q):
+            """
+            The integrator will try to solve this equation
+            at each time step.
+            Note: t is the integration time and is non-monotonic.
+            """
+            # Set the joint values, without checking the joint limits
             robot.SetActiveDOFValues(q, CheckLimitsAction.Nothing)
+
             return fn_vectorfield()
 
         def fn_status_callback(t, q):
+            """
+            Check joint-limits and collisions for a specific joint
+            configuration. This is called multiple times at DOF
+            resolution in order to check along the entire length of the
+            trajectory.
+            Note: Currently this is called after each integration time
+            step, which means we are doing more checks than required.
```

Comment:
This comment is not true. The `fn_callback` function below is called after every integration timestep, but only calls `fn_status_callback` if the robot has moved more than DOF resolution. If that is not the case, then it is a bug - possibly because of the comment about `GetCollisionCheckPts` below.